src/app/directives/image/image.directive.ts

```diff
@@ -158,18 +158,19 @@ export class AuthenticatedImageDirective implements OnChanges {
    * Returns true if all other image options are exhausted
    */
   private use404Image(): boolean {
-    return this.images.count() === 0 && this.currentImage === this.defaultImage;
+    return (this.images.count() === 0 && this.currentImage === this.defaultImage);
   }
 
   /**
    * Returns true if the default image is the only option left available
    */
   private useDefaultImage(): boolean {
-    return (
+    const useDefaultImage = (
       this.images.count() === 0 &&
-      this.currentImage !== this.defaultImage &&
-      !!this.defaultImage
+      this.currentImage !== this.defaultImage
```

Comment:
explain this change. The previous code seems more correct. Why did you change it?

Reply:
This is an artifact of my testing. It has since been removed